js/scripts/zone_loaders/village_zone_loaders.js

```diff
@@ -0,0 +1,84 @@
+const loadVillageCanyon = () => {
+    const queueAssets = () => {
+
+    };
+
+    const addEntities = () => {
+
+    };
+
+    queueAssets();
+    ASSET_MGR.downloadAll(addEntities);
+};
+
+const loadVillageField = () => {
+    const queueAssets = () => {
+
+    };
+
+    const addEntities = () => {
+
+    };
+
+    queueAssets();
+    ASSET_MGR.downloadAll(addEntities);
+};
+
+const loadVillageInsideCave = () => {
+    const queueAssets = () => {
+
+    };
+
+    const addEntities = () => {
+
+    };
+
+    queueAssets();
+    ASSET_MGR.downloadAll(addEntities);
+};
+
+const loadVillageMain = () => {
+    const queueAssets = () => {
+        ASSET_MGR.queueDownload(PapaChad.SPRITESHEET);
+        ASSET_MGR.queueDownload(Block.SPRITESHEET);
+    };
+
+    const addEntities = () => {
+        // Add a layer of blocks to the floor.
+        for (let x = ZONE.MIN_BLOCK.x; x <= ZONE.MAX_BLOCK.x; x++) {
+            GAME.addEntity(new Block(new Vector(x, ZONE.MAX_BLOCK.y), Block.DIRT));
+        }
+        // Place chad above the blocks.
+        const blockPos = new Vector(5, 20);
```

Comment:
Got me thinking. When we load a new zone we will want to set CHAD's new coordinates to the inverse of his last known x coordinates from the previous zone. We will want the y to be the same. Not sure how hard that could be if we are trashing the contents of the previous zone when loading a new one. Am I missing something or is that correct? Perhaps we want to have a class that stores CHAD's last coordinates from the previous zone. @B1GD3VD0G @tclaar @NathanHinthorne

Reply:
I think your conclusion about using inverses is false, because it falsely assumes that all dimensions are the same size. However, I've already thought about this. Haven't done it yet, but prob was going to tomorrow. We'll need to keep track of LAST_ZONE, which we will change every time we switch zones. Then, in the zone loader, we will just need to place chad at a certain point, based on the value of LAST_ZONE (of which there will only be a handful of possibilities for any given zone)

Reply:
Good point on there being different sized zones. I appreciate the thought you put into this. Looking forward!